### PR TITLE
ekf2: enable initialization without baro data

### PR DESCRIPTION
**Pull request by bresch**
- Pass initialization when no baro data is available and when baro fusion is disabled (`EKF2_BARO_CTRL` = 0)
- Add fake height fusion when no vertical aiding is available

TODO:
- [ ] review vertical position validity flag

### CODE REVIEW

File: src/modules/ekf2/EKF/fake_height_control.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		ECL_INFO("stop fake height fusion");
+		_using_synthetic_hgt = false;
+
+		resetEstimatorAidStatus(_aid_src_fake_pos);
+	}
+}
+
+void Ekf::fuseFakeHgt()
+{
+	const float obs_var = sq(0.5f);
```
Comment (priseborough) [inline src/modules/ekf2/EKF/fake_height_control.cpp:114]: We should use the existing baro noise parameter for this to facilitate tuning for different use cases, eg rover vs boat in swells

**Review by priseborough — COMMENTED** [review]
All good from my review, but we should make the observation noise tuneable which could be done using the baro parameter. We would need to update parameter tuning notes to make this clear.

**Comment by bresch** [comment]
@priseborough We apparently already have a parameter for this: https://github.com/PX4/PX4-Autopilot/pull/20021/commits/8d1c38d7555ed78302d653502a44d2dd420a2c0a

**Review by priseborough — DISMISSED** [review]

**Comment by dagar** [comment]
Rebased to resolve merge conflict (csv auto commit) and it's now broken due to `_time_last_imu` usage. I'll update the branch.

File: src/modules/ekf2/EKF/fake_height_control.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	P.uncorrelateCovarianceSetVariance<1>(9, sq(_params.pos_noaid_noise));
+}
+
+void Ekf::stopFakeHgtFusion()
+{
+	if (_using_synthetic_hgt) {
+		ECL_INFO("stop fake height fusion");
+		_using_synthetic_hgt = false;
+
+		resetEstimatorAidStatus(_aid_src_fake_pos);
```
Comment (dagar) [inline src/modules/ekf2/EKF/fake_height_control.cpp:108]: This is problematic because it's going to reset things for x & y as well.

**Review by dagar — CHANGES_REQUESTED** [review]
The sharing of `_aid_src_fake_pos` is potentially problematic. Not a big deal for fake pos/hgt, but we should still try to get it right.

Personally I would either split _aid_src_fake_pos for pos and hgt, or do a bit more to merge the routines (like merging fuseFakeHgt() and fuseFakePosition(). You could still have separate state machines for enabling/disable fake pos/hgt, but ultimately the fusion and updating _aid_src_fake_pos would be centralized and happen together. 

Keeping fake height completely separate also seems fine if that's what you'd prefer.

File: src/modules/ekf2/EKF/ekf.h [hunk 3]
```diff
@@ -466,6 +466,7 @@ class Ekf final : public EstimatorInterface
 	uint64_t _time_prev_gps_us{0};	///< time stamp of previous GPS data retrieved from the buffer (uSec)
 	uint64_t _time_last_aiding{0};	///< amount of time we have been doing inertial only deadreckoning (uSec)
 	bool _using_synthetic_position{false};	///< true if we are using a synthetic position to constrain drift
+	bool _using_synthetic_hgt{false};	///< true if we are using a synthetic height to constrain drift
```
Comment (dagar) [inline src/modules/ekf2/EKF/ekf.h:469]: Any reason we don't expose these as control status flags?

**Comment by bresch** [comment]
@dagar 
- The "fake position" and "fake height" are now completely separate. Even if it sounds "nice" to have them together, it makes the logic less readable and less "standard".
- The en-/disabled flags for fake pos/hgt are now part of `_control_status.flags`
- The baro sample is popped out the baro buffer in `controlBaroHeightFusion`, used in place and **not** stored as `_baro_sample_delayed` anymore.
- Baro data isn't used in `initialiseFilter` anymore

**Review by dagar — APPROVED** [review]

**Pull request merged**